### PR TITLE
KM261 ✅ Add ArgoCD upgrade

### PR DESCRIPTION
**Pull request by yngvark**
## Description

Add upgrade for upgrading ArgoCD to from 1.7.2 to 2.1.7.

## Motivation and Context

[KM261](https://trello.com/c/tNKlCgY1/261-upgrade-argocd)

## How to prove the effect of this PR?

1. Optional, if argocd version isn't 1.7.2: In an existing cluster, set the ArgoCD deployment image to 1.7.2 (deployment name: `argocd-application-controller`).
1. git clone this repository and checkout this PR's branch
1. `cd 0.0.81.argocd && go build`
1. Run this upgrade by following the steps described in the section of the root README which was updated by this PR

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: 0.0.81.argocd/pkg/argocd/api.go [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+
+func (a ArgoCD) partiallyRemoveArgoCD(services *core.Services, clusterID api.ID) error {
+	if a.dryRun {
+		return nil
+	}
+
+	a.log.Info("Removing Old ArgoCD")
+
+	a.log.Debug("Deleting Helm Release")
+	err := services.Helm.DeleteHelmRelease(context.Background(), client.DeleteHelmReleaseOpts{
```
Comment (deifyed) [inline 0.0.81.argocd/pkg/argocd/api.go:139]: By removing ArgoCD first, the user will lose their applied argocd-application.yamls

File: 0.0.81.argocd/pkg/argocd/api.go [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+
+func (a ArgoCD) preflight() error {
+	isInstalled, err := a.kubectl.hasDeployment(argoCDNamespace, argoCDDeploymentName)
+	if err != nil {
+		return fmt.Errorf("getting argocd: %w", err)
+	}
+
+	if !isInstalled {
+		a.log.Info("ArgoCD is not installed, not doing anything")
+		return nil
```
Comment (deifyed) [inline 0.0.81.argocd/pkg/argocd/api.go:101]: We should probably abort the upgrade when ArgoCD is missing. User probably set `integrations.argoCD: false`

File: 0.0.81.argocd/upgrade.go [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+func upgrade(context Context, flags cmdFlags) error {
+	opts := argocdPkg.Opts{
+		DryRun:  flags.dryRun,
+		Confirm: flags.confirm,
+	}
+
+	argocd, err := argocdPkg.New(context.log, opts)
+	if err != nil {
+		return fmt.Errorf("craeting argocd: %w", err)
```
Comment (deifyed) [inline 0.0.81.argocd/upgrade.go:16]: *creating :D

**Review by deifyed — APPROVED** [review]
Nice! I dig the readability of the ArgoCD.Upgrade() func

File: 0.0.81.argocd/pkg/argocd/api.go (hunk 2 above)
Comment (yngvark) [reply 0.0.81.argocd/pkg/argocd/api.go:101]: Ops, that's a bug, good catch. Fixed.

**Pull request merged**